include/ccmath/math/nearest/nearbyint.hpp

```diff
@@ -8,7 +8,64 @@
 
 #pragma once
 
+#include <ccmath/math/compare/isinf.hpp>
+#include <ccmath/math/compare/isnan.hpp>
+#include <ccmath/math/compare/signbit.hpp>
+#include <ccmath/math/nearest/trunc.hpp>
+#include <type_traits>
+
 namespace ccm
 {
+	/**
+	 * @brief The nearest integer value to num, according to the rounding mode FE_TONEAREST, is returned.
+	 * @tparam T The type of the number.
+	 * @param num A floating-point value.
+	 * @return If no errors occur, the rounded floating point value. Otherwise, returns the input floating point value unmodified.
+	 */
+	template <class T, std::enable_if_t<std::is_floating_point_v<T>, bool> = true>
+	constexpr T nearbyint(T num) noexcept
+	{
+		// If num is ±∞ or ±0, num is returned, unmodified.
+		if (ccm::isnan(num) || ccm::isinf(num) || num == T(0)) { return num; }
+
+		T int_part	= ccm::trunc(num);
+		T frac_part = num - int_part;
+		if (frac_part == T(0.0)) { return int_part; }
+		if (frac_part > T(0.5) || (frac_part == T(0.5) && static_cast<long long>(int_part) % 2 != 0)) { return int_part + T(1); }
+		if (frac_part < T(-0.5) || (frac_part == T(-0.5) && static_cast<long long>(int_part) % 2 != 0)) { return int_part - T(1); }
+		return int_part;
+	}
+
+	/**
+	 * @brief The nearest integer value to num, according to the rounding mode FE_TONEAREST, is returned.
+	 * @tparam Integer The type of the number.
+	 * @param num An integral value.
+	 * @return If no errors occur, the rounded floating point value. Otherwise, returns the input integer value unmodified.
+	 */
+	template <class Integer, std::enable_if_t<std::is_integral_v<Integer>, bool> = true>
+	constexpr double nearbyint(Integer num) noexcept
+	{
+		return static_cast<double>(num);
+	}
+
+	/**
+	 * @brief The nearest integer value to num, according to the rounding mode FE_TONEAREST, is returned.
+	 * @param num A float value.
+	 * @return If no errors occur, the rounded floating point value. Otherwise, returns the input integer value unmodified.
+	 */
+	constexpr float nearbyintf(float num)
+	{
+		return nearbyint(num);
+	}
+
+	/**
+	 * @brief The nearest integer value to num, according to the rounding mode FE_TONEAREST, is returned.
+	 * @param num A long double value.
+	 * @return If no errors occur, the rounded floating point value. Otherwise, returns the input integer value unmodified.
+	 */
+	constexpr long double nearbyintl(long double num)
+	{
+		return nearbyint(num);
```

Comment:
Please apply the full qualified namespace of the function. This is mostly to be a proactive measure to avoid possible issues with compilers getting confused. Should not happen, but we have no clue what the end users environment will look like so better safe than getting a bug report later.

Reply:
Clarification question. Do you mean change the function to be something like:

```cpp
constexpr long double nearbyintl(long double num) {
    return ccm::nearbyint(num);
 }
 ```

Reply:
Yes that is what I want. Makes things more explicit. Generally when calling functions that could theoretically have a collision I'm generally just explicit with it. ^^